packages/devtools_app/lib/src/screens/debugger/program_explorer_controller.dart

```diff
@@ -105,18 +109,34 @@ class ProgramExplorerController extends DisposableController
     ScriptRef script,
     List<VMServiceObjectNode> nodes,
   ) {
+    final searchCondition = (node) => node.script?.uri == script.uri;
     for (final node in nodes) {
-      final result = node.firstChildWithCondition(
-        (node) => node.script?.uri == script.uri,
-      );
+      final result = node.firstChildWithCondition(searchCondition);
       if (result != null) {
         selectNode(result);
         result.expandAscending();
+        _selectedNodeIndex.value = _calculateNodeIndex(searchCondition);
         return;
       }
     }
   }
 
+  int _calculateNodeIndex(
+      bool matchingNodeCondition(VMServiceObjectNode node)) {
+    var index = 0;
+    for (final node in _rootObjectNodes.value) {
+      final matchingNode = depthFirstTraversal(
+        node,
+        returnCondition: matchingNodeCondition,
+        exploreChildrenCondition: (node) => node.isExpanded,
+        action: (_) => index++,
+      );
+      if (matchingNode != null) return index;
+    }
+    // If the node wasn't found, return 0.
+    return 0;
```

Comment:
maybe we should return -1, and then prevent scrolling the program explorer if the selected node index is -1? Otherwise, we'll jump the scroll view around because we couldn't find the index.

Reply:
Good point! Done